### PR TITLE
Feat:Cart

### PR DESCRIPTION
**Pull request by Shrinkhlarajpoot**
Added Various Features in Cart 
-Add item in Cart
-Remove Item from Cart
-Increase and Decrease Quantity in a Cart
-Receipt with all the payment calculations.

### CODE REVIEW

File: src/frontend/components/MainCard/MainCard.js [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
+  const [inCart,setInCart]=useState(false)
+  useEffect(()=>{
+ cart.cartProducts.find((prod)=>prod._id === products._id) && setInCart(true)
+ 
+  
+
+  },[cart.cartProducts])
+
+  const addToCart = async() => {
+    console.log("add to cart")
```
Comment (AbhishekkGautam) [inline src/frontend/components/MainCard/MainCard.js:20]: We should remove the console.log

File: src/frontend/pages/CartScreen/components/Cartdetails.js [hunk 2]
```diff
@@ -24,7 +43,7 @@ const Cartdetails = () => {
         <div className="details">
           <div className="details__content">
             <h3>Total MRP</h3>
-            <h3>Rs 4000</h3>
+            <h3>Rs {totalPriceofitems - totalPriceofitems * (8 / 100)}</h3>
```
Comment (AbhishekkGautam) [inline src/frontend/pages/CartScreen/components/Cartdetails.js:46]: We can take out this operation and put it inside a function ( helper ).

**Review by AbhishekkGautam — APPROVED** [review]
There are many console.log in different files. we should remove them before commit. 
Everything else looks good. Clean & structured code. 👍

File: src/frontend/components/MainCard/MainCard.js (hunk 1 above)
Comment (Shrinkhlarajpoot) [reply src/frontend/components/MainCard/MainCard.js:20]: Thanks for mentioning...Will surely take care of it.

File: src/frontend/pages/CartScreen/components/Cartdetails.js (hunk 2 above)
Comment (Shrinkhlarajpoot) [reply src/frontend/pages/CartScreen/components/Cartdetails.js:46]: got it.thanks

File: src/frontend/components/MainCard/MainCard.js [hunk 3]
```diff
@@ -1,6 +1,35 @@
-import { Navigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { Navigate, useNavigate } from "react-router-dom";
+import { useAuth } from "../../context/authContext";
+import { useCart } from "../../context/cartContext";
```
Comment (msdeshmukh009) [inline src/frontend/components/MainCard/MainCard.js:4]: Barrel export will be useful here.

File: src/frontend/components/MainCard/MainCard.js [hunk 4]
```diff
@@ -1,6 +1,35 @@
...
+import { addToCartService } from "../../services";
 import "./MainCard.css";
 const MainCard = ({ products }) => {
+  const navigate = useNavigate();
+  const { auth } = useAuth();
+  const { cart, setCart } = useCart();
+  const [inCart,setInCart]=useState(false)
+  useEffect(()=>{
```
Comment (msdeshmukh009) [inline src/frontend/components/MainCard/MainCard.js:12]: Maybe you don't need to have an effect here.

File: src/frontend/utils/Sumofpriceandlength.jsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+const Sumofprice = (array) =>{
+  return array.reduce((accum, initial) => (accum = accum +(initial.qty * initial.price)), 0);
+}
+
+
+// const Sumofpricewithdiscount = (array) =>{
+//   const totalprice = array.reduce((accum, initial) => (accum = accum + (initial.qty * initial.price)), 0);
+//   return totalprice-(8/100)*totalprice
+// }
```
Comment (msdeshmukh009) [inline src/frontend/utils/Sumofpriceandlength.jsx:14]: You can remove this comment if you don't need this.

**Review by msdeshmukh009 — APPROVED** [review]
This is looking good. Good to merge.

**Pull request merged**